planner/core/logical_plan_builder.go

```diff
@@ -580,28 +580,55 @@ func (b *PlanBuilder) buildProjectionField(id, position int, field *ast.SelectFi
 func (b *PlanBuilder) buildProjection(p LogicalPlan, fields []*ast.SelectField, mapper map[*ast.AggregateFuncExpr]int) (LogicalPlan, int, error) {
 	b.optFlag |= flagEliminateProjection
 	b.curClause = fieldList
-	proj := LogicalProjection{Exprs: make([]expression.Expression, 0, len(fields))}.init(b.ctx)
-	schema := expression.NewSchema(make([]*expression.Column, 0, len(fields))...)
-	oldLen := 0
-	for _, field := range fields {
-		newExpr, np, err := b.rewrite(field.Expr, p, mapper, true)
-		if err != nil {
-			return nil, 0, errors.Trace(err)
+	var remainingFields []*ast.SelectField
+	fieldSet := make([]bool, len(fields))
+	for idx, field := range fields {
+		if strings.Contains(field.Text(), ast.Ifnull) {
+			if fnExpr, ok := field.Expr.(*ast.FuncCallExpr); ok {
+				if cnExpr, ok := fnExpr.Args[0].(*ast.ColumnNameExpr); ok {
+					var colFound *model.ColumnInfo
+					if ds, ok := p.(*DataSource); ok {
+						colFound = model.FindColumnInfo(ds.Columns, cnExpr.Name.Name.L)
+						if mysql.HasNotNullFlag(colFound.Flag) {
```

Comment:
Can we extract these judgement to a function? So that it has strong scalability.